crates/rpc/src/method/get_block_with_txs.rs

```diff
@@ -85,7 +85,7 @@ pub async fn get_block_with_txs(context: RpcContext, input: Input) -> Result<Out
         let transactions = transaction
             .transactions_for_block(header.number.into())
             .context("Reading transactions from database")?
-            .context("Transaction data missing")?
+            .unwrap_or(vec![])
```

Comment:
Maybe this is something we could fix in the underlying `query_transactions_by_block()` call instead? That now has an explicit check for "no transactions" and just returns `None`:

```rust
        let mut rows = stmt.query(params![&block_number])?;
        let Some(row) = rows.next()? else {
            return Ok(None);
        };
```

One could argue that we could maybe remove the `Option` and just return an empty vector if we consider blocks with no transactions valid.

Otherwise we'd still have the exact same issue in other JSON-RPC method calls that fetch the transactions for a block (like `starknet_traceTransaction` and `starknet_traceBlockTransactions`).

Reply:
Understood, thanks for this detailed feedback! I have the made the changes as per this, can you take a look?